connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/WorkerCoordinator.java

```diff
@@ -397,4 +391,160 @@ private String ownerUrl(String connector) {
 
     }
 
+    public static class ConnectorsAndTasks {
+        public static final ConnectorsAndTasks EMPTY =
+                new ConnectorsAndTasks(Collections.emptyList(), Collections.emptyList());
+
+        private final Collection<String> connectors;
+        private final Collection<ConnectorTaskId> tasks;
+
+        private ConnectorsAndTasks(Collection<String> connectors, Collection<ConnectorTaskId> tasks) {
+            this.connectors = connectors;
+            this.tasks = tasks;
+        }
+
+        public static ConnectorsAndTasks copy(Collection<String> connectors,
+                                              Collection<ConnectorTaskId> tasks) {
+            return new ConnectorsAndTasks(new ArrayList<>(connectors), new ArrayList<>(tasks));
+        }
+
+        public static ConnectorsAndTasks embed(Collection<String> connectors,
```

Comment:
nit: i found this terminology a bit weird. do we (or something else) use `embed` like this normally? If not using a builder pattern but wanting a static method instead of public constructor to build it (which, tbh, seems like a fine solution to me...), I would expect `build(..., ...)` or similar.